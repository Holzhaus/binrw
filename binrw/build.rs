use std::env;

fn main() {
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_SEARCH");
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_LIB");
    println!("cargo:rerun-if-changed=build.rs");

    if cfg!(bintrace) {
        if let Ok(link_search) = env::var("BINTRACE_LINK_SEARCH") {
            println!("cargo:rustc-link-search={}", link_search)
        }

        if let Ok(link_lib) = env::var("BINTRACE_LINK_LIB") {
            println!("cargo:rustc-link-lib={}", link_lib);
        } else {
            println!("cargo:warning=No bintrace backend provided");
        }
    }
}
