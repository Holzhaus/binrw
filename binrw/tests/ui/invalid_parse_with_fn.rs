#![feature(generic_associated_types)]
use binrw::BinRead;

#[derive(BinRead)]
struct Foo {
    #[br(parse_with = does_not_exist)]
    a: i32,
}

fn main() {}
