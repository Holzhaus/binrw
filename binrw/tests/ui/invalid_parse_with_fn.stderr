error[E0425]: cannot find value `does_not_exist` in this scope
 --> $DIR/invalid_parse_with_fn.rs:6:23
  |
6 |     #[br(parse_with = does_not_exist)]
  |                       ^^^^^^^^^^^^^^ not found in this scope
